service_grpc_flight/src/lib.rs

```diff
@@ -746,33 +736,34 @@ fn cmd_from_descriptor(flight_descriptor: FlightDescriptor) -> Result<FlightSQLC
 /// FlightSQL is using the new header names.
 fn get_flightsql_namespace(metadata: &MetadataMap) -> Result<String> {
     let mut found_header_keys: Vec<String> = vec![];
+
     for key in IOX_FLIGHT_SQL_DATABASE_HEADERS {
         if metadata.contains_key(key) {
             found_header_keys.push(key.to_string());
         }
     }
-    if found_header_keys.len() > 1 {
-        return TooManyFlightSQLDatabasesSnafu {
-            header_names: found_header_keys,
-        }
-        .fail();
+
+    if found_header_keys.is_empty() {
+        return NoFlightSQLDatabaseSnafu.fail();
     }
 
-    if let Some(v) = metadata.get(IOX_FLIGHT_SQL_DATABASE_HEADERS[0]) {
-        let v = v.to_str().context(InvalidDatabaseHeaderSnafu)?;
-        return Ok(v.to_string());
-    } else if let Some(v) = metadata.get(IOX_FLIGHT_SQL_DATABASE_HEADERS[1]) {
-        let v = v.to_str().context(InvalidDatabaseHeaderSnafu)?;
-        return Ok(v.to_string());
-    } else if let Some(v) = metadata.get(IOX_FLIGHT_SQL_DATABASE_HEADERS[2]) {
-        let v = v.to_str().context(InvalidDatabaseHeaderSnafu)?;
-        return Ok(v.to_string());
-    } else if let Some(v) = metadata.get(IOX_FLIGHT_SQL_DATABASE_HEADERS[3]) {
-        let v = v.to_str().context(InvalidNamespaceHeaderSnafu)?;
-        return Ok(v.to_string());
+    // if all the keys specify the same database name, return the name
+    let mut database_name: &str = "";
```

Comment:
I think if you wanted to be slightly more  Rust idomatic, you could use an Option here instead of empty string

so like

```suggestion
    let mut database_name = None;
```

And then below instead of

```rust
 if database_name.is_empty() {
                database_name = v;
            } else if database_name != v {
```

Something like

```
 if database_name.is_none() {
                database_name = Some(v);
  } else if let Some(database_name) = database_name.as_ref() {
     if database_name != v {
      // return error
    }
}
```


```

Reply:
However since "" is not a valid database name I think this code will work as well